connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorCheckpointTask.java

```diff
@@ -103,10 +113,11 @@ public void start(Map<String, String> props) {
         targetAdminClient = config.forwardingAdmin(config.targetAdminConfig("checkpoint-target-admin"));
         metrics = config.metrics();
         idleConsumerGroupsOffset = new HashMap<>();
-        checkpointsPerConsumerGroup = new HashMap<>();
+        Optional<Map<String, Map<TopicPartition, Checkpoint>>> checkpoints = readCheckpoints(config);
```

Comment:
This is a potentially long blocking operation, and those should be avoided in start() methods because while the task is starting, it can't be stopped, and if the task can't be stopped within `task.shutdown.graceful.timeout.ms` it is aggressively cancelled.

Since the main thread needs the result from readCheckpoints, I think it would be fine to check if it's been loaded and if not, just return an empty poll().

Reply:
Thanks for the explanation, and suggestion. We'll take another look at reading the checkpoint topic asynchronously from the start() method.